### ISSUE TITLE
`inlinestrings` fails on `Union{Missing, String}` with length-1 strings

### ISSUE THREAD

**Issue by nilshg** [JuliaStrings/InlineStrings.jl#34]
Another length-1 issue:

```
julia> using InlineStrings, Missings

julia> inlinestrings([missing, "e"])
ERROR: MethodError: no method matching Union{}(::String)
Closest candidates are:
  Union{}(::Any...) at boot.jl:261
  (::Type{T})(::Rational{S}) where {S, T<:AbstractFloat} at rational.jl:114
  (::Type{T})(::Real, ::RoundingMode) where T<:AbstractFloat at rounding.jl:200
  ...
Stacktrace:
 [1] (::Core.TypeofBottom)(a::String)
   @ Core .\boot.jl:261
 [2] _inlinestrings(itr::Vector{Union{Missing, String}}, st::Int64, #unused#::Type{Missing}, IS::Base.HasShape{1}, res::Vector{Missing}, i::Int64)
   @ InlineStrings C:\Users\ngudat\.julia\packages\InlineStrings\7BGwu\src\InlineStrings.jl:886
 [3] inlinestrings(itr::Vector{Union{Missing, String}})
   @ InlineStrings C:\Users\ngudat\.julia\packages\InlineStrings\7BGwu\src\InlineStrings.jl:869
 [4] top-level scope
   @ REPL[8]:1

julia> inlinestrings([missing, "ee"])
2-element Vector{Union{Missing, String3}}:
 missing
 "ee"

(jl_Yp2qxZ) pkg> st
Status `C:\Users\ngudat\AppData\Local\Temp\jl_Yp2qxZ\Project.toml`
  [842dd82b] InlineStrings v1.1.3
```

**Comment by nickrobinson251** [comment]
Thanks for another great bug report!

Hopefully fixed by #35

**Comment by nilshg** [comment]
Thanks for the quick turnaround - trying your branch locally works for me, look forward to the merge & patch release!
